### ISSUE TITLE
[Ray] Changing the logging directory through `ray start` doesn't work

### ISSUE THREAD

**Issue by shrekris-anyscale** [ray-project/ray#40628]
### What happened + What you expected to happen

A user reported that changing the logging directory to something other than `/tmp` with `ray start` doesn’t work because of two issues:

* It only changes the directory on the head node. The worker nodes still use the `/tmp` directory
* Larger scale Ray workloads crash with some sort of error. The user didn’t have the error handy.

### Versions / Dependencies

Ray 2.7.1

### Reproduction script

The user didn't have a reproduction handy.

### Issue Severity

High: It blocks me from completing my task.

**Comment by rkooo567** [comment]
Potentially related to https://github.com/ray-project/ray/issues/42623

**Comment by tianshihan818** [comment]
I guess the user used the option `--temp-dir` in `ray start` to change the temp dir path of the worker node. If so, the reason why only the head node could change but the worker node could not is that, the option `--temp-dir` will be ignored when starting worker node. See the [source code](https://github.com/ray-project/ray/blob/fd56d10ef697df411c54e92169eb55a9d5ae3de1/python/ray/scripts/scripts.py#L625).

And in this case, the worker node get the `ray_params` with the `temp_dir = None`, then when [initializing the worker node](https://github.com/ray-project/ray/blob/fd56d10ef697df411c54e92169eb55a9d5ae3de1/python/ray/scripts/scripts.py#L928), it will call a function `self._init_temp()` to get the temp dir. For the worker node, as the `ray_params.temp_dir` is set to be None before, it will always get the temp dir from the head node gsc k-v store, see the [source code](https://github.com/ray-project/ray/blob/3627e946dca7dd90b9f99dd6b3641910b10f932e/python/ray/_private/node.py#L418). Finally, the temp dir path of worker nodes will be same as the head node.

To change the temp dir of the worker node directly, you can export an environment var `RAY_TMPDIR`. For example:
```bash
# start head node
ray start --head --temp-dir /tmp/ray-head ...
or 
export RAY_TMPDIR=/tmp/ray-head && ray satrt --head ...
# start worker node
export RAY_TMPDIR=/tmp/ray-worker
ray start --address $head_ip
```

**Comment by rkooo567** [comment]
yeah it was intentional that worker node cannot set temp_dir (since there's no use case, and it complicates deployment). But given this use case, this approach makes sense
